src/cryptsetup/cryptsetup.c

```diff
@@ -1073,6 +1100,21 @@ static int attach_luks_or_plain_or_bitlk_by_tpm2(
                         if (r != -EAGAIN) /* EAGAIN means: no tpm2 chip found */
                                 return r;
                 } else {
+                        r = attach_luks2_by_tpm2(cd, name, flags);
+                        /* EAGAIN  means: no tpm2 chip found
+                         * ENOTSUP means: no libcryptsetup plugins support */
+                        if (!IN_SET(r, -ENOTSUP, -EAGAIN)) {
+                                if (r == -ENXIO)
+                                        return log_debug_errno(SYNTHETIC_ERRNO(EAGAIN),
+                                                               "No TPM2 metadata matching the current system state found in LUKS2 header, falling back to traditional unlocking.");
+                                if (r == -ENOENT)
+                                        return log_debug_errno(SYNTHETIC_ERRNO(EAGAIN),
+                                                               "No TPM2 metadata enrolled in LUKS2 header, falling back to traditional unlocking.");
+                                return r;
+                        }
+                }
+
+                if (r != -EAGAIN) {
```

Comment:
nit: it would be more immediately obvious that this branch is a fallback, if the check was `r == -ENOTSUP`